gse-util/src/main/java/com/powsybl/gse/util/GroovyCodeEditor.java

```diff
@@ -129,6 +147,30 @@ public GroovyCodeEditor(Scene scene) {
         codeArea.setOnDragOver(this::onDragOver);
         codeArea.setOnDragDropped(this::onDragDropped);
         codeArea.setOnSelectionDrag(p -> allowedDrag = true);
+
+        codeArea.caretPositionProperty().addListener((observable, oldvalue, newvalue) -> {
+            searchingForMatches = false;
+            try {
+                if (oldvalue != null && oldvalue > 0 && tokenIsBracket(codeArea.getText(oldvalue - 1, oldvalue))) {
+                    codeArea.setStyleSpans(0, computeHighlighting(codeArea.getText()));
+                }
+            } catch (IndexOutOfBoundsException ex) {
```

Comment:
Is there other way to detect oldvalue doesn't exists anymore than catching an exception?

Reply:
updated